src/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,321 @@
+import React from "react";
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {
+  testStyledSystemPadding,
+  testStyledSystemWidth,
+} from "../../__spec_helper__/__internal__/test-utils";
+import CarbonProvider from "../carbon-provider";
+import Form from "../form";
+
+import Sidebar, { SidebarProps } from ".";
+
+const AllTheProviders = ({ children }: { children: React.ReactNode }) => (
+  <CarbonProvider validationRedesignOptIn>{children}</CarbonProvider>
+);
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+
+afterEach(() => {
+  jest.runOnlyPendingTimers();
+  jest.useRealTimers();
+});
+
+test("sidebar element has necessary dialog properties", () => {
```

Comment:
```suggestion
test("sidebar element has 'aria-modal' attribute set to 'true'", () => {
```

**suggestion** (non-blocking): I prefer this more explicit wording since we're only asserting against that single property. I guess you left it generic in case we later add more, but I'd prefer to only change the description at that point while testing for more properties (or alternatively to add a new, equally explicit, test). I think it's more obvious what the test is doing, both when reading the code and when seeing the test description if it happens to fail, when it's explicit.

It would also make it more clear what the difference is between this and the following test, where setting `enableBackgroundUI` makes `aria-modal` have the value "false", and where you've been explicit about what's happening in that one.